vendor/mpdf/mpdf/src/Tag/P.php

```diff
@@ -1,9 +1,25 @@
 <?php
-
+ 
 namespace Mpdf\Tag;
+ 
+class P extends InlineTag
+{	
+ 
+	public function open($attr, &$ahtml, &$ihtml)
+	{	
 
-class P extends BlockTag
-{
+		parent::open($attr, $ahtml, $ihtml);
```

Comment:
This change the behavior of P elements for everybody, is that a correct interpretation?
Then you probably fixed this problem by creating an problem at another place?

Another problem is that the library has to be fixed manually on every update of the library. So is there not an alternative approach that solves the issue in a manner that does not need manual fixes of the library each update?

Reply:
Correct. This will make paragraphs inline, as it should be in HTML. That makes pictures float.

 I've being using it for a year, works fine on my setup, but sure it requires more different tests. The most questionable thing is that we replace a natural paragraph line break with a BR element which has different stylesheet. For example, slightly bigger top/bottom margins. So, it is a workaround, but it makes things work. 

I wish there were something better than patching mpdf. Well... may be we can patch "p" in runtime from the plugin? Can PHP replace a class constructor?

Reply:
I do no understand why mpdf would do this, and therefore i do also not know if this change is right.
Changes for mpdf library should be proposed in the https://github.com/mpdf/mpdf repository.